parser: fix CREATE SEQUENCE ... AS invalid_type

otan opened the pull request:
Resolves #82192

Release note (bug fix): Fix a panic in CREATE SEQUENCE ... AS ... if the
type is not a valid integer type.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82213)
<!-- Reviewable:end -->

otan committed:
parser: fix CREATE SEQUENCE ... AS invalid_type

Release note (bug fix): Fix a panic in CREATE SEQUENCE ... AS ... if the
type is not a valid integer type.

mgartner commented on pkg/sql/parser/sql.y:
```diff
@@ -8124,6 +8124,12 @@ sequence_option_elem:
+                                  // Ideally, we should actually do type resolution here and print out the invalid
+                                  // type name, but we don't do type resolution in the parser at the moment.
+                                  if parsedType == nil {
+                                      sqllex.Error(fmt.Sprintf("invalid integer type in CREATE SEQUENCE"))
```
nit: match PG's message and error code: 

```
marcus=# create sequence s1 as abc;
ERROR:  42704: type "abc" does not exist
LINE 1: create sequence s1 as abc;
```
